### PR TITLE
Add location normalizer service

### PR DESCRIPTION
**Pull request by oleg1std**

### CODE REVIEW

File: Api/Controllers/LocationsController.cs [hunk 1]
```diff
@@ -105,7 +106,17 @@ public async Task<IActionResult> GetLastModifiedDate()
         }
 
 
+        [ProducesResponseType(typeof(DateTime), (int) HttpStatusCode.OK)]
```
Comment (mingazhev) [inline Api/Controllers/LocationsController.cs:109]: This is not true, nor the response type nor the status code

File: Api/Infrastructure/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -403,6 +404,9 @@ public static IServiceCollection AddServices(this IServiceCollection services)
 
             services.AddTransient<INetstormingResponseService, NetstormingResponseService>();
             services.AddTransient<IBookingWebhookResponseService, BookingWebhookResponseService>();
+
+            services.AddNameNormalizationServices();
+            services.AddSingleton<ILocationNormalizer, LocationNormalizer>();
```
Comment (mingazhev) [inline Api/Infrastructure/ServiceCollectionExtensions.cs:409]: It depends on scoped service - EdoContext. Does this work?

File: Api/Services/Locations/LocationNormalizer.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        private void Normalize(List<Location> locations)
+        {
+            foreach (var location in locations)
+            {
+                Normalize(location);
+            }
+        }
+
+
+        void NormalizeCountry(Location location)
```
Comment (mingazhev) [inline Api/Services/Locations/LocationNormalizer.cs:55]: Need to add access modifiers there and further, or move to local functions

File: Api/Infrastructure/Logging/LoggerEvents.cs [hunk 4]
```diff
@@ -32,5 +32,6 @@ public enum LoggerEvents
         AgentAuthorizationFailure = 1111,
         CounterpartyStateAuthorizationSuccess = 1120,
         CounterpartyStateAuthorizationFailure = 1121,
+        LocationNormalized = 1122
```
Comment (mingazhev) [inline Api/Infrastructure/Logging/LoggerEvents.cs:35]: I suggest to add a spacing between logger events which are far from each other, e.g. 1130.
In this case we could add new event ids right after 'CounterpartyStateAuthorizationFailure'

File: Api/Controllers/LocationsController.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Controllers/LocationsController.cs:109]: And method should be documented

File: Api/Infrastructure/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (oleg1std) [reply Api/Infrastructure/ServiceCollectionExtensions.cs:409]: Yes, it works. But, I agree, it is a mismatch,

**Pull request merged**